hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java

```diff
@@ -192,20 +192,17 @@ private EntityManagerFactoryBuilderImpl(
 
 		this.persistenceUnit = persistenceUnit;
 
-		if ( integrationSettings == null ) {
-			integrationSettings = Collections.emptyMap();
-		}
+		final Map integtrationSettingsCopy = integrationSettings == null ? new HashMap() : new HashMap( integrationSettings );
```

Comment:
We should avoid `Collections.emptyMap` if there is possibility to modify it later.
Also, I do think defensive copying never hurts (a good practise as per 'Effective Java') so I create one up front. That also simplified the logic later on.